compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2194,14 +2162,23 @@ public void visit(BLangXMLAccessExpr xmlAccessExpr) {
 
     @Override
     public void visit(BLangTypedescExpr accessExpr) {
-        BIRVariableDcl tempVarDcl =
-                new BIRVariableDcl(accessExpr.getBType(), this.env.nextLocalVarId(names), VarScope.FUNCTION,
-                                   VarKind.TEMP);
-        this.env.enclFunc.localVars.add(tempVarDcl);
-        BIROperand toVarRef = new BIROperand(tempVarDcl);
-        setScopeAndEmit(new BIRNonTerminator.NewTypeDesc(accessExpr.pos, toVarRef, accessExpr.resolvedType,
-                                                         Collections.emptyList()));
-        this.env.targetOperand = toVarRef;
+        createNewTypedescInst(accessExpr.pos, accessExpr.getBType(), accessExpr.resolvedType);
+    }
+
+    private boolean checkAndSetRecordTypedesc(BLangExpression expression) {
+        if (expression.getKind() != NodeKind.TYPEDESC_EXPRESSION) {
+            return false;
+        }
+
+        BLangTypedescExpr typedescExpr = (BLangTypedescExpr) expression;
+        BType resolvedType = typedescExpr.resolvedType;
+
+        if (resolvedType.tag == TypeTags.RECORD && getTypedescVariable(resolvedType) != null) {
+            this.env.targetOperand = new BIROperand(getTypedescVariable(resolvedType));
```

Comment:
Shall we make `getTypedescVariable(resolvedType)` as a variable, because we calculate it twice in here